src/core/reactor.cc

```diff
@@ -1874,9 +1875,13 @@ reactor::open_file_dma(std::string_view nameref, open_flags flags, file_open_opt
                     ::ioctl(fd, XFS_IOC_FSSETXATTR, &attr);
                 }
             }
+            r = ::fstat(fd, &st);
+            if (r == -1) {
+                return wrap_syscall(r, st);
+            }
```

Comment:
Thought experiment: with io_uring this translates to IOURING_OPEN_AT + IOSQE_FIXED_FILE, followed by IORING_OP_STATX + IOSQE_FIXED_FILE. They can then be done by the same io_uring_enter call (though likely with a workqueue to help). /cc  @tchaikov

Reply:
(it would have been nice of io_uring to be able to say, take the fd from the linked openat call)

Reply:
IO uring has this file descriptors 'pre-registration' which helps with reusing fd's across calls. Are we using it?

Reply:
(https://man7.org/linux/man-pages/man3/io_uring_register_files.3.html )

Reply:
That's what IOSQE_FIXED_FILE uses (IIUC). We're not using it yet.